### PR TITLE
Improve ephemeral error logging

### PR DESCRIPTION
**Pull request by joshblum**
once this is in should help pinpoint the root cause of your error @mmaxim

### CODE REVIEW

File: go/ephemeral/user_ek_box_storage.go [hunk 1]
```diff
@@ -109,7 +109,8 @@ func (s *UserEKBoxStorage) Get(ctx context.Context, generation keybase1.EkGenera
 
 	defer s.Unlock() // release the lock after we unbox
 	if cacheItem.HasError() {
-		return userEK, cacheItem.Error()
+		err = cacheItem.Error()
```
Comment (mmaxim) [inline go/ephemeral/user_ek_box_storage.go]: What is going on here?

File: go/ephemeral/user_ek_box_storage.go [hunk 2]
```diff
@@ -230,7 +231,8 @@ func (s *UserEKBoxStorage) unbox(ctx context.Context, userEKGeneration keybase1.
 		s.G().Log.CDebugf(ctx, "unable to get from deviceEKStorage %v", err)
 		switch err.(type) {
 		case erasablekv.UnboxError:
-			return userEK, newEKUnboxErr(UserEKStr, userEKGeneration, DeviceEKStr, userEKBoxed.DeviceEkGeneration)
+			err = newEKUnboxErr(UserEKStr, userEKGeneration, DeviceEKStr, userEKBoxed.DeviceEkGeneration)
```
Comment (mmaxim) [inline go/ephemeral/user_ek_box_storage.go]: And here?

File: go/ephemeral/user_ek_box_storage.go (hunk 1 above)
Comment (joshblum) [reply go/ephemeral/user_ek_box_storage.go]: this will allow `err` to hold the correct value for logs in the `Trace`

Comment (mmaxim) [reply go/ephemeral/user_ek_box_storage.go]: If you return it, `err` gets set, so I don't think you need to do this.

**Pull request merged**